src/Library/demos/Launcher/main.blp

```diff
@@ -0,0 +1,94 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.StatusPage {
+  title: _("Launcher");
+  description: _("Collect the arguments that are needed to open a File or URI");
+
+  Adw.Clamp {
+    maximum-size: 640;
+    Box {
+      orientation: vertical;
+      spacing: 24;
+
+      Gtk.StackSwitcher {
+        stack: stack;
+        halign: center;
+      }
+
+      Gtk.Stack stack {
+        transition-type: crossfade;
+        vexpand: true;
+
+        Gtk.StackPage {
+          name: "file_launcher";
+          title: _("File Launcher");
+          child:
+            Box {
+              orientation: vertical;
+              halign: center;
+              spacing: 24;
+              Box {
+                halign: center;
+                Button launch_file {
+                  label: _("Launch");
+                  styles ["suggested-action", "pill"]
+                }
+              }
+
+              Box {
+                styles ["linked"]
+
+                Button file_details {
+                  label: _("Open File");
+                }
+                Button file_location {
+                  label: _("Open Folder");
+                }
+              }
```

Comment:
```suggestion
              Button file_location {
                label: _("Open Folder");
              }
```

Removing the "Open File", which conflicts with the meaning of the "Launch" button. The "Open Folder" folder button is a good idea, but I think it should be somehow connected or associate with the "Launch" button; any ideas?

Reply:
I thought a workaround for this; I changed the open file to file name instead so that it doesnt conflict with launch button; I had another idea how about renaming this to File details wherein i log the details of the file such as size, name etc

Reply:
I'm just not sure what purpose it serves, and it seems a bit confusing that it is presented with the same importance as "Open Folder".

"Launch" and "Open Folder" both represent methods of the `Gtk.FileLauncher` class, while the other button simply prints the file name to the console. Maybe instead the file selection button could have a caption with the current file name?

Reply:
Here's what I would do:

* Move the constructors for both launchers to their respective sections in the code view.
* Connect to `notify::file` and query the file name to set a caption on the "Choose another file" button
* Set the default file and URI after the signals are connected, so they populate at first run
* Make an equally prominent button for "View"

Here is what I'd imagine that looking like:

[Screencast from 2023-08-05 16-52-36.webm](https://github.com/sonnyp/Workbench/assets/1265208/b6e910d8-d40a-48fb-a487-52cc932adbbd)